### PR TITLE
Participant Selector Load Performance 

### PR DESCRIPTION
**Pull request by AndersonTarren**
# Critical Changes

# Changes
Improves the performance of load and search within the Participant Selector component.

# Issues Closed

# New Metadata

# Deleted Metadata

# Definition of Done
  Refer to [Asteroids DoD document](https://salesforce.quip.com/iq2mAy4i62oM) to see any additional details for the items below
- [ ] Any net new LWC work has Sa11y tests & 50% or above lines JEST test coverage  
- [ ] CRUD/FLS is enforced in Apex
- [ ] Permission sets are updated to account for CRUD|FLS|Tab|Classes
- [ ] Field sets are updated to account for new fields
- [x] UX approval or UX not necessary
- [x] Link the pull request and work item by PR comment and Chatter post respectively, e.g. GUS: W-0000000: Work Name (https://github.com/SalesforceFoundation/PMM/pull/303)
- [ ] All **acceptance criteria** have been met
    - [x] Developer
    - [ ] Code Reviewer
    - [ ] QA
- [x] PR contains draft release notes
- [ ] QE story level testing completed

### CODE REVIEW

**Comment by AndersonTarren** [comment]
[W-11550867](https://gus.lightning.force.com/a07EE000013IY5QYAW)

File: force-app/main/default/classes/ServiceScheduleCreatorController.cls [hunk 1]
```diff
@@ -47,9 +47,23 @@ public with sharing class ServiceScheduleCreatorController {
...
+    public static SelectParticipantModel getSelectParticipantModel(
+        Id serviceId,
+        String searchText,
+        String cohortId
+    ) {
+        System.debug(cohortId);
+        System.debug('Controller');
+        Id cohortIdConvert = null;
+        if (cohortId != null && cohortId != '') {
+            cohortIdConvert = (Id) cohortId;
```
Comment (jjbennett) [inline force-app/main/default/classes/ServiceScheduleCreatorController.cls:59]: Nitpicky but automatic casting will happen when you call the service method.

File: force-app/main/default/labels/CustomLabels.labels-meta.xml [hunk 2]
```diff
@@ -1096,4 +1096,11 @@
         <shortDescription>This feature is only available on the web application</shortDescription>
         <value>This feature is only available on the web application.</value>
     </labels>
+    <labels>
+        <fullName>Too_Many_Participants</fullName>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Too many results returned from participant selector</shortDescription>
+        <value>More than 1000 results found, refine by searching or filtering by Program Cohort</value>
```
Comment (jjbennett) [inline force-app/main/default/labels/CustomLabels.labels-meta.xml:1104]: Did we run this past Farzana yet?

File: force-app/main/default/lwc/participantSelector/participantSelector.html [hunk 3]
```diff
@@ -76,6 +76,15 @@
                                             onclick={handleSelectAll}
                                         ></lightning-button>
                                     </lightning-layout-item>
+                                    <lightning-layout-item
+                                        size="12"
+                                        if:true={show1kMessage}
+                                    >
+                                        <c-scoped-notification
+                                            theme="warning"
```
Comment (jjbennett) [inline force-app/main/default/lwc/participantSelector/participantSelector.html:84]: Are we using info over in CaseMan, is warning too strong?

**Review by jjbennett — COMMENTED** [review]
A couple minor comments but I also notice that when I have more than 3 letters in the search tearm delete until there are only 3 left and then delete again, it deletes all 3 remaining letters. Let me know if you need a video for reference.

File: force-app/main/default/classes/ProgramEngagementSelector_TEST.cls [hunk 4]
```diff
@@ -112,6 +114,121 @@ public with sharing class ProgramEngagementSelector_TEST {
             'Expected that both the actual and expected list are associated to the same program cohort'
         );
     }
+    //Tarren Start
```
[the remark this answers is not in the document]
Comment (AndersonTarren) [reply force-app/main/default/classes/ProgramEngagementSelector_TEST.cls:117]: (facepalm)

File: force-app/main/default/labels/CustomLabels.labels-meta.xml (hunk 2 above)
Comment (AndersonTarren) [reply force-app/main/default/labels/CustomLabels.labels-meta.xml:1104]: Yes, she's reviewing the text today.

**Review by fanwar-salesforce — CHANGES_REQUESTED** [review]
Please change the message to:

This Program has more than 1,000 contacts. Refine the results by filtering by Program Cohort.

**Comment by AndersonTarren** [comment]
@fanwar-salesforce  - Question on the message.  The proposed change doesn't mention the ability to refine their results by using the text search.  Also, the returned result is Program Engagements, so mentioning 1000 Contacts might be confusing.

**Review by jjbennett — CHANGES_REQUESTED** [review]
We found an issue with SOSL where only 250 records are being returned. This work item has been implemented as required but we will need to create a new work item to determine how we want to handle the 250 record limit.

**Review by fanwar-salesforce — CHANGES_REQUESTED** [review]
Here we go for the value: 

Program has more than 1,000 participants. Refine the results by searching or filtering by Program Cohort.

**Comment by AndersonTarren** [comment]
[W-11550966](https://gus.lightning.force.com/a07EE000013J1eVYAS)
[W-11801818](https://gus.lightning.force.com/a07EE0000175aZOYAY)

**Review by jjbennett — APPROVED** [review]

**Pull request merged**